core/textile/utils/utils.go

```diff
@@ -94,7 +95,8 @@ func GetThreadContext(parentCtx context.Context, threadName string, dbID thread.
 	}
 
 	var publicKey crypto.PubKey
-	if publicKey, err = kc.GetStoredPublicKey(); err != nil {
+	var privKey crypto.PrivKey
+	if privKey, publicKey, err = kc.GetStoredKeyPairInLibP2PFormat(); err != nil {
```

Comment:
Is this really needed? Because this means it will ask for the keychain pass again

Reply:
Yeah it fails without this.  Because the token in the context could be from a previous hub ctx e.g. this fails with invalid thread token if we don't create a new token.  And to create the new token it expects a privkey. It's related to the last bug in the change log.

Reply:
Makes sense, but this will become really annoying on OSX 😅 
I think we need to include a cache in `keychain.go` so that it pulls/stores the private key from memory.

I can take a stab at this later in the afternoon if you agree with that.

Reply:
Fair enough, I'll keep this PR open until we add that then @dmerrill6. I can also take a stab at it on this branch.  If so I'm thinking of adding the privkey as a field on the `kc` instance that's gets set on startup. And a helper function to return that value. Does anything else need to be done or you had something else in mind?

Reply:
That sounds good. I think the current getter (`GetStoredKeyPairInLibP2PFormat`) can look if the value is set and return it, or do the current flow otherwise. Also `GenerateKeyPair, GenerateKeyFromMnemonic, GenerateKeyPairWithForce, ImportExistingKeyPair, DeleteKeypair` need to clear this cache.

Reply:
@dmerrill6 please review the following commit to address this https://github.com/FleekHQ/space-daemon/pull/189/commits/caa6f8bee2c20a29d2c88cfe795b28c0cd0a34f5.

* `GetStoredKeyPairInLibP2PFormat` like you said checks for the cache and returns that, if not found sets it
* Added the save to cache in `generateAndStoreKeyPair` so it handles all of `GenerateKeyPair`, `GenerateKeyPairWithForce` and `GenerateKeyFromMnemonic`
* Update/remove cache in `ImportExistingKeyPair`, `DeleteKeypair`
* Assuming `GetStoredPublicKey` was working without pw prompt so didn't touch that

Reply:
Yea GetStoredPublicKey shouldn't change. The changes LGTM 👍